app/controllers/users/cart_products_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Users::CartProductsController < Users::ApplicationController
+  before_action :set_cart_product, only: %i[destroy]
+
+  def index
+    @cart_products = current_user.cart_products.map
+  end
+
+  def create
+    @cart_product = current_user.cart_products.build(cart_product_params)
+    if @cart_product.save
+      redirect_to users_cart_products_url, notice: 'CartProduct was successfully created.'
+    else
+      render [:users, @cart_product.product]
+    end
+  end
+
+  def destroy
+    @cart_product.destroy
```

Comment:
失敗しても false 返して進んでしまうので、 `!` をつけてエラーが起きたら気づけるようにした方が良いです。

```suggestion
    @cart_product.destroy!
```